cmd/cluster-agent/subcommands/start/command.go

```diff
@@ -477,3 +493,26 @@ func initializeRemoteConfig(ctx context.Context) (*rcclient.Client, *rcservice.S
 
 	return rcClient, rcService, nil
 }
+
+func registerChecks() {
+	// Required checks
+	registerCheck(true, cpu.CheckName, cpu.New)
+	registerCheck(true, memory.CheckName, memory.New)
+	registerCheck(true, uptime.CheckName, uptime.New)
+	registerCheck(true, io.CheckName, io.New)
+	registerCheck(true, filehandles.CheckName, filehandles.New)
+
+	// Flavor specific checks
+	registerCheck(kubernetesapiserver.Enabled, kubernetesapiserver.CheckName, kubernetesapiserver.New)
```

Comment:
Would it work if we renamed `init` to `Init`, and simply called it from here? Would remove the need to rename and make public check names, which would make this PR a lot smaller.

Reply:
Eventually the check loader will be a component, and no longer globally accessible from within each check. So we'd either need to add the loader as an argument to every init method, or refactor the init methods much like this PR already does (to return a check instance that we can then pass to the loader). 

So I think while we could simply rename the init methods, removing the usage of globals (the loader) helps us down the road with further componentization.

Reply:
> Eventually (soon) the check loader will be a component, and no longer globally accessible from within each check. So we'd either need to add the loader as an argument to every init method,

Adding a parameter sounds entirely reasonable to me. Also, IIRC there was interest to make each individual check (factory) into a component. `Init` that takes other components as arguments would seem to fit nicely into component constructor role.

Reply:
> @ogaca-dd what do you think? - do you feel that one approach fits the component model better

I think so but this is not in the roadmap.